apps/web/src/state/buyCrypto/hooks.ts

```diff
@@ -0,0 +1,246 @@
+import { useTranslation } from '@pancakeswap/localization'
+import { ChainId, Currency } from '@pancakeswap/sdk'
+import { useAtom, useAtomValue } from 'jotai'
+import { useRouter } from 'next/router'
+import { ParsedUrlQuery } from 'querystring'
+import { useCallback, useEffect } from 'react'
+import { BuyCryptoState, buyCryptoReducerAtom } from 'state/buyCrypto/reducer'
+import { useAccount } from 'wagmi'
+import toString from 'lodash/toString'
+import { useActiveChainId } from 'hooks/useActiveChainId'
+import formatLocaleNumber from 'utils/formatLocaleNumber'
+import ceil from 'lodash/ceil'
+
+import { Field, replaceBuyCryptoState, selectCurrency, setMinAmount, setUsersIpAddress, typeInput } from './actions'
+
+type CurrencyLimits = {
+  code: string
+  maxBuyAmount: number
+  minBuyAmount: number
+}
+
+export function useBuyCryptoState() {
+  return useAtomValue(buyCryptoReducerAtom)
+}
+
+function parseTokenAmountURLParameter(urlParam: any): string {
+  return typeof urlParam === 'string' && !Number.isNaN(parseFloat(urlParam)) ? urlParam : ''
+}
+
+export const fetchMinimumBuyAmount = async (
+  inputCurrencyId: string,
+  outputCurrencyId: string,
+): Promise<{ [curr: string]: CurrencyLimits }> => {
+  try {
+    const response = await fetch(
+      `https://api.moonpay.com/v3/currencies/${outputCurrencyId.toLowerCase()}/limits?apiKey=pk_live_Ch5fat39X8NvMZwih2k7hK4sDrKanSPz&baseCurrencyCode=${inputCurrencyId.toLowerCase()}&areFeesIncluded=true`,
```

Comment:
Make all the endpoint `https://api.moonpay.com/v3` in [here](https://github.com/pancakeswap/pancake-frontend/blob/develop/apps/web/src/config/constants/endpoints.ts)